edgelet/edgelet-http/src/lib.rs

```diff
@@ -217,9 +223,37 @@ where
 
         let protocol = Arc::new(protocol);
 
+        let (lock, limit) = if let ConcurrencyThrottling::Limited(limit) = concurrency {
```

Comment:
Maybe I am wrong, but it seems like you would only need the 'lock' as it contains the limit itself. So I guess just using the initial value could work for where you need that?

Reply:
I need it slightly below. Lock give you a way to access the initial value.